mount/mount_windows.go

```diff
@@ -101,5 +163,30 @@ func Unmount(mount string, flags int) error {
 
 // UnmountAll unmounts from the provided path
 func UnmountAll(mount string, flags int) error {
+	if mount == "" {
+		// This isn't an error, per the EINVAL handling in the Linux version
+		return nil
+	}
+
 	return Unmount(mount, flags)
 }
+
+func (m *Mount) bindMount(target string) error {
+	for _, option := range m.Options {
+		if option == "ro" {
+			return errors.Wrap(ErrNotImplementOnWindows, "read-only bind mount")
+		}
+	}
+
+	if err := os.Remove(target); err != nil {
+		return err
+	}
+
+	// TODO: We don't honour the Read-Only flag.
+	// It's possible that Windows simply lacks this.
+	return os.Symlink(m.Source, target)
```

Comment:
Having just found https://github.com/golang/go/issues/26033, I'm suspicious that this being a symlink will go wrong when exposed to `filepath.Walk` and friends, and bind-mounts should be proper directory junction points instead. (They'll still go wrong, but at least it's _wrong_ wrong, and more-consistent with a `bind` mount on Linux). I still can't make it read-only though.

`filepath.Walk` called _on_ the symlink does the right thing, so it's a question of "what should a bind-mount do when walking the directory that contains it?"

Reply:
I need to look into this, per https://github.com/containerd/continuity/issues/174 I suspect `filepath.Walk` also does not do the right thing when called on junction points (symlinks or mountpoints) directly. If I'm wrong now and was right in 2020, then perhaps I can resolve that in the test suite after all.

Reply:
So, I've been playing around with the Bind Filter [mentioned in this comment](https://github.com/containerd/containerd/pull/4419#discussion_r975460692). Apparently we can use it to create "bind" mounts or even mount volumes. It does not seem to use reparse points, so the issue of walk not visiting symlinks may be fixed if we switch to that instead of using ```SetVolumeMountPoint()``` or creating directory junctions. Will try it out with your branch and if it works, I'll create a PR against this branch.

Reply:
Opened a PR that makes use of the bind filter here: https://github.com/TBBle/containerd/pull/1

This should allow us to have proper "views" without hacks as well as proper "bind" mounts, using the same API.